go/vt/vttablet/onlineddl/executor.go

```diff
@@ -2826,6 +2889,23 @@ func (e *Executor) reviewRunningMigrations(ctx context.Context) (countRunnning i
 		postponeCompletion := row.AsBool("postpone_completion", false)
 		elapsedSeconds := row.AsInt64("elapsed_seconds", 0)
 
+		if stowawayTable := row.AsString("stowaway_table", ""); stowawayTable != "" {
+			// whoa
+			// stowawayTable is an original table stowed away while cutting over a vrepl migration, see call to cutOverVReplMigration() down below in this function.
+			// In a normal operation, the table should not exist outside the scope of cutOverVReplMigration
+			// If it exists, that means a tablet crashed while running a cut-over, and left the database in a bad state, where the migrated table does not exist.
+			// thankfully, we have tracked this situation and just realized what happened. Now, first thing to do is to restore the original table.
+			log.Infof("found stowaway table %s in migration %s. Restoring it as %s", stowawayTable, uuid, onlineDDL.Table)
+			if _, err := e.renameTableIfApplicable(ctx, stowawayTable, onlineDDL.Table); err != nil {
+				// unable to restore table; we bail out, and we will try again next round.
+				return countRunnning, cancellable, err
+			}
+			// OK good, table restored. We can remove the record.
```

Comment:
By design `renameTableIfApplicable()` doesn't return an err if the original or stowaway tables don't exist. So is it ok to expect that we have actually restored the table if it reaches here? If neither or both of the tables exist, we will not get an error even though the function didn't actually rename the tables. Possibly this is an impossible situation, but wanted to check.

Reply:
This is indeed intentional, which is why I named the function "ifApplicable". There is indeed a situation where this can happen:

Two lines below, after we've renamed the tables, we remove the `stowaway_table` value from the migration:

https://github.com/vitessio/vitess/pull/9832/files#diff-059c9f46e8d270d9c5514ef2b08679035eb0daaa8d95074e34ef43a81d50dc37R2904-R2906

However, if the tablet crashes just after renaming the table (back) and updating the fact that the table has been renamed -- then next time it runs -- or next tablet that runs -- sees a `stowaway_table` and thinks it still needs to do the rename. This is why I made the function idempotent. It OK to try it again and it will return with no error if there is nothing to rename.